modyn/trainer_server/internal/dataset/online_dataset.py

```diff
@@ -270,13 +280,20 @@ def _get_data(
             callback()
 
     def _get_transformed_data_tuple(
-        self, key: int, sample: memoryview, label: int, weight: float | None
+        self, key: int, sample: memoryview, label: int | None = None, weight: float | None = None
     ) -> tuple | None:
         assert self._uses_weights is not None
         self._sw.start("transform", resume=True)
         # mypy complains here because _transform has unknown type, which is ok
         transformed_sample = self._transform(sample)  # type: ignore
         self._sw.stop("transform")
+
+        if not self._include_labels:
+            if self._uses_weights:
+                return key, transformed_sample, weight
+            return key, transformed_sample
```

Comment:
shouldnt for consistency be the label returned as None? i am not sure, but _get_data_from_storage you use None, while here the semantics change. Could you elaborate a bit here? Can we somehow compact those different cases?